### ISSUE TITLE
Use a logger with different logging levels

### ISSUE THREAD

**Issue by mgechev** [guess-js/guess#165]
Currently, inside of the Guess.js webpack plugin there are a lot of statements in the form:

```js
if (this._config.debug) {
  console.warn(...);
}
```

Instead of having this duplication of `this._config.debug` checks, it'd be better if we isolate the logic into a logger with different logging levels. We can either create a custom logger (no dependencies, smaller payload) or go with an existing one (code reuse).

**Comment by vprasanth** [comment]
I could take a look if you'd like.

**Comment by mgechev** [comment]
Thanks! Assigned it to you. What approach would you prefer? Custom logger or something from npm?

**Comment by vprasanth** [comment]
I can try custom if you want to keep the size/dependencies minimal and usage simple. How do you see the log level being set? Through env, explicitly on the logger itself, or through some other mechanism?

**Comment by mgechev** [comment]
Custom sounds good. Currently, the Guess.js plugin accepts a `debug` flag. Two log levels controlled by the flag sounds good enough unless you have other ideas.

**Comment by vprasanth** [comment]
I'll start there :)

**Comment by mgechev** [comment]
@vprasanth let me know if you want to discuss this further :)

**Comment by vprasanth** [comment]
Hey, sorry for the delay, was at a cottage this weekend :) 

Anyways, I ended up adding a couple more log levels -- but set the logger to `DEBUG`. I added the additional levels to cover `error` and `warn`. So in total there is `DEBUG`, `INFO`, `WARN`, `ERROR`, and `OFF`. Default for the logger is set to `INFO`.

I've so far only updated the `prefetch-aot-plugin` file,  I noticed some other places within the guess-webpack package that use `console`, wasn't sure if you wanted those updated as well. Figured I'd wait for a review then update the rest if wanted. Let me know :)
